src/Shape.ts

```diff
@@ -0,0 +1,54 @@
+import { Point } from './Point';
+
+export abstract class Shape {
+  protected readonly defaultColor: string = 'green';
+  protected readonly defaultFilledStatus: boolean = true;
+
+  protected color: string = '';
+  protected filled: boolean = false;
+  protected points: Point[] = [];
+
+  static validate(points: Point[]) {
+    if (points.length < 3) {
+      throw new Error('Invalid Shape. Must have at least 3 points.');
+    }
+  }
+
+  constructor(points: Point[], color?: string, filled?: boolean) {
```

Comment:
Same here. Please, implement overloaded constructor, instead of using default parameters.

Reply:
Constructor override applied